packages/adapter-prisma/src/index.ts

```diff
@@ -15,114 +15,130 @@
  *
  * @module @auth/prisma-adapter
  */
-import type { PrismaClient, Prisma } from "@prisma/client"
+import type { PrismaClient, Prisma } from "@prisma/client";
 import type {
   Adapter,
   AdapterAccount,
   AdapterSession,
   AdapterUser,
-} from "@auth/core/adapters"
+} from "@auth/core/adapters";
 
 export function PrismaAdapter(
-  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>
+  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>,
+  options?: {
+    accountModel?: string;
+    authenticatorModel?: string;
+    sessionModel?: string;
+    userModel?: string;
+    verificationTokenModel?: string;
+  }
 ): Adapter {
-  const p = prisma as PrismaClient
+  const p = prisma as PrismaClient;
+  const accountModel = p[options?.accountModel ?? "account"];
+  const authenticatorModel = p[options?.authenticatorModel ?? "authenticator"];
+  const sessionModel = p[options?.sessionModel ?? "session"];
+  const userModel = p[options?.userModel ?? "user"];
```

Comment:
Looks like there's some TS issues here. See: https://github.com/nextauthjs/next-auth/actions/runs/12449775424/job/34755845781?pr=12401

Reply:
Not sure how to set up TS locally correctly, having a bunch of errors namely all args of methods have implicit any but I am not overriding the default tsconfig file afaik 🤔 
<img width="586" alt="Screenshot 2024-12-22 at 23 50 04" src="https://github.com/user-attachments/assets/7185b9b9-c65e-4a6b-92f6-46f279906e3e" />